tests/cxx/binary_type_trait.cc

```diff
@@ -0,0 +1,48 @@
+//===--- binary_type_trait.cc - test input file for iwyu ------------------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+#include "tests/cxx/binary_type_trait-d1.h"
+
+int main() {
+    // IWYU: BinaryTypeTraitBase is defined in "tests/cxx/binary_type_trait-i1.h", which isn't directly #included.
```

Comment:
Please use the pattern-matching form of these diagnostic assertions, see e.g. `funcptrs.cc` for examples.